packages/babel-plugin-proposal-pattern-matching/src/index.js

```diff
@@ -0,0 +1,316 @@
+        });
+
+        return arrayTest;
+      case "ObjectMatchPattern":
+        objectTest = makeIsObjectTest(id);
+
+        pattern.children.forEach(propPattern => {
+          const subPropTest = makeTest(path, id, propPattern, defines, false);
+
+          objectTest = t.logicalExpression("&&", objectTest, subPropTest);
+        });
+
+        if (pattern.restIdentifier !== null) {
+          const exists_key = pattern.children.map(
+            propPattern => propPattern.key.name,
+          );
+
+          // make a new object
+          // iterate the matched key
+          // check if the matched key in the old objects
+          // if it's not, add it to the new object
+          //
+          // let $restID = {};
+          //
+          // for (let $key in id) {
+          //  if ($key !== $key1 && $key !== $key2 && ...) {
+          //    newobj[key] = id[key];
+          //  }
+          // }
+          if (exists_key.length > 0) {
+            defines.push(
+              t.variableDeclaration("var", [
+                t.variableDeclarator(
+                  pattern.restIdentifier,
+                  t.objectExpression([]),
+                ),
+              ]),
+            );
+            key_id = path.scope.generateUidIdentifier("key");
+
+            defines.push(
+              template(`
+              for (var KEY_ID in ID) {
+                if (REST_TEST) {
+                  REST_ID[KEY_ID] = ID[KEY_ID];
+                }
+              }
+              `)({
+                KEY_ID: key_id,
+                REST_TEST: makeRestTest(key_id, exists_key),
+                REST_ID: pattern.restIdentifier,
+                ID: id,
+              }),
+            );
+          } else {
+            // exists_key.length === 0
+            defines.push(
+              template(`
+              var REST_ID = Object.assign({}, ID);
+              `)({
+                REST_ID: pattern.restIdentifier,
+                ID: id,
+              }),
+            );
+          }
+        }
+
+        return objectTest;
+      case "ObjectPropertyMatchPattern":
+        objectPropTest = t.callExpression(
+          t.memberExpression(id, t.identifier("hasOwnProperty"), false),
+          [t.stringLiteral(pattern.key.name)],
+        );
+
+        newId = t.memberExpression(id, pattern.key, false);
+        if (pattern.value === null) {
+          pattern.value = pattern.key;
+        }
+        objPropSubTest = makeTest(path, newId, pattern.value, defines, false);
+
+        objectPropTest = t.logicalExpression(
+          "&&",
+          objectPropTest,
+          objPropSubTest,
+        );
+
+        return objectPropTest;
+      case "Identifier":
+        if (isRoot) {
+          if (pattern.name === "Array") {
+            return template.expression(`Array.isArray(ID)`)({
+              ID: id,
+            });
+          } else {
+            return template.expression(`(PATTERN[Symbol.match] &&
```

Comment:
The proposal refers to Symbol.matches, not Symbol.match

Reply:
@ricochet1k Thx. I am going to fix it.